im From c4d13c5e820c94cfec6da68bd58d61ff863d9bd8 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Thu, 12 May 2022 21:32:21 -0400
Subject: [PATCH] Use -Wno-unused-but-set-variable unconditionally on *nix

This flag is supported by Clang >=13

xref: https://releases.llvm.org/13.0.0/tools/clang/docs/ReleaseNotes.html#new-compiler-flags
---
 mypyc/build.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/mypyc/build.py b/mypyc/build.py
index f5ff0201f..ca427c52f 100644
--- a/mypyc/build.py
+++ b/mypyc/build.py
@@ -518,10 +518,8 @@ def mypycify(
             '-Werror', '-Wno-unused-function', '-Wno-unused-label',
             '-Wno-unreachable-code', '-Wno-unused-variable',
             '-Wno-unused-command-line-argument', '-Wno-unknown-warning-option',
+            '-Wno-unused-but-set-variable'
         ]
-        if 'gcc' in compiler.compiler[0] or 'gnu-cc' in compiler.compiler[0]:
-            # This flag is needed for gcc but does not exist on clang.
-            cflags += ['-Wno-unused-but-set-variable']
     elif compiler.compiler_type == 'msvc':
         # msvc doesn't have levels, '/O2' is full and '/Od' is disable
         if opt_level == '0':
-- 
2.33.1

